### PR TITLE
DOC: Update the documetation of homepage of website

### LINKED ISSUE #14754: Better pointer to dev docs on website
It was very difficult to find the developer's guide on the matplotlib website. It would be nice to have a pointer to https://matplotlib.org/devel/index.html on the homepage/somewhere easy to fine

### PR DESCRIPTION
**Pull request by sameshl**
It was very difficult to find the developer's guide on the matplotlib website.
So added a pointer to https://matplotlib.org/devel/index.html on the homepage which makes it easy to find.

closes #14754

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [x] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

**Comment by story645** [comment]
Hi, looking at the page that gets built (https://23060-1385122-gh.circle-artifacts.com/0/home/circleci/project/doc/build/html/index.html), looks like the change didn't stick. This is what I see:
image

**Comment by timhoffm** [comment]
The whole main page could use an overhaul. See also #13382.

**Comment by sameshl** [comment]
@timhoffm Yeah. The main page looks a bit wordy and tough to navigate. I would love to continue and work on https://github.com/matplotlib/matplotlib/issues/13382!
